Moved from our smsLibrary to the central smsLibrary

kummy98 opened the pull request:
## Premise

Moved from our smsLibrary to the central smsLibrary.
I still not removed the old library from the project, i only removed from the dependencies of the app.

## Changes

-All the methods of the old library were replaced with the method's of the new one.
-Added a new class that manage all the message received from the app(what our alarmAndLocateResponseActivity used to do).

## Side Changes

- Changed the way some specs were wrote

Commit:
-Moved to Central library from smsLibrary of our group1
-The directory of the old library it's still in the project.

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/ServiceManager.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5052095&bid=16633313)

turcato-niccolo reviewed (dismissed):
Ok, I've briefly tried it and looks like it works. The code is good and actually this improved

ricdezen commented on app/build.gradle:
```diff
@@ -24,14 +24,14 @@ android {
     implementation 'androidx.room:room-runtime:2.2.3'
     implementation 'com.github.rehacktive:waspdb:1.1.1'
     implementation "androidx.fragment:fragment:1.2.1"
+    implementation 'com.github.Cogno-IDU:smslibrary:v2.2'
```
I know we said the "common" library, but we also specified our fork of the library, otherwise where do all our random generators go?

RicDeZen commented:
I have a couple points to make.

1. I believed we said to use OUR fork of the library, because our random peer generator has been approved but not merged, so half our tests cannot run on the common library.
2. You did not check the above issue by running the tests or you would have noticed.
3. Why didn't you pull before opening this? It's normal for conflicts to arise on a pr that is already open if another gets merged first, but the changes that are giving these conflicts were already in place before you opened this.

turcato-niccolo commented:
> 2\. You did not check the above issue by running the tests or you would have noticed.

I didn't do that either 😱

Commit:
Merge branch 'development' of https://github.com/RicDeZen/ingsw-group1-findmyphone into Findmyphone_commonlibrary

# Conflicts:
#	app/build.gradle
#	app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java
#	app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java

Commit:
-Changes made after the review of Mr.De zen
-Resolved all the conflicts
-Checked also that all the tests work
-There are some errors with the test class RandomSMSLogEventGeneratorTest

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/ServiceManager.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5052095&bid=16633313)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/event/SMSLogEvent.java:
```diff
@@ -3,6 +3,8 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 
+import com.eis.smslibrary.SMSPeer;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - com.eis.smslibrary.SMSPeer.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5052095)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/event/SMSLogEvent.java:
```diff
@@ -3,6 +3,8 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 
+import com.eis.smslibrary.SMSPeer;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'com.eis.smslibrary.SMSPeer'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5052095)

turcato-niccolo replied:
There are these useful checks on android studio, so that you can avoid these things
![image](https://user-images.githubusercontent.com/56435959/75160777-f0445e00-571a-11ea-9a99-c2cdb0e5989a.png)

kummy98 replied:
Thank you, i didn't know about that

Commit:
-Removed an unused import

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/ServiceManager.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5052095&bid=16633313)

ricdezen reviewed (commented):
I test partono, la richiesta di squillo parte, la richiesta di posizione parte e torna indietro. Dovrebbe essere tutto a posto. Do un'ultima controllata al codice in se.

RicDeZen commented:
Ah ma se non mergio questa col piffero che parte la Android CI giustamente
